devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java

```diff
@@ -125,6 +126,15 @@
     @Parameter(property = "uberJar", defaultValue = "false")
     private boolean uberJar;
 
+    @Parameter(defaultValue = "${project.groupId}", required = true, readonly = true)
+    private String group;
```

Comment:
This part - and the `BuildInfo` stuff and the `DevMojo` stuff - I think @aloubyansky should comment on as it seems to overlap with the bootstrapping branch, at least in concept.

Reply:
I think it might be better to introduce a *global* configuration for the application name and version, as a config rather than a Mojo property.  The application name could default to the artifact ID.

Reply:
So would you like me to introduce something like `ApplicationConfig` as a build time configuration object?

Reply:
Yeah as a separate PR maybe?  One question is whether the output name should have a group ID, which is a bit of a loaded question because it implies more questions about whether/how the output artifact(s) can/should be published.

Reply:
FWIW, Spring only defines `spring.application.name` (see https://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html) and from what I have seen people just supply a name, no group.

So would you prefer I first create a PR introducing `ApplicationConfig` and then rebase this PR onto `master` when it lands?

Reply:
Sure that sounds good.  Note that the app config should not have any opinions about the kind/structure of output or the type of image being produced; it probably could even just be the app name and version TBH.

Reply:
Sounds good! No I just need to figure out how to create a Config with default values (I'll ping you if I need any help with that but I would rather do my own digging first since it's a good chance to see parts of the code I haven't touched yet).

Reply:
Hopefully you'll find it very simple. :)  I'll be around in any case.